Change to only warn for custom Function

albanD opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32940 Change to only warn for custom Function**
* #32839 Clean views created inside no_grad that are modified inplace
* #32790 Add allow_rebase_history flag and fix codegen functions for multiple views

As discussed with @gchanan, we do a deprecation cycle for this as some users might rely on the current behavior.
But given that their custom backward is not called in this case (or called with partial gradients), we agree that this is not the expected behavior and will disallow it.

Differential Revision: [D19814115](https://our.internmc.facebook.com/intern/diff/D19814115)

albanD committed:
Change to only warn for custom Function

[ghstack-poisoned]

albanD committed:
Update on "Change to only warn for custom Function"




[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit f5186480:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=f5186480ad4fba3a49fc98d4abcca79e2db05ae1).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4404989) pytorch_windows_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=94782006) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=94782006))
<details>
<summary>
<code>C:\Users\circleci\project\torch/csrc/autograd/variable.h(391): error C2062: type 'unknown-type' unexpected</code>
</summary>

```
CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION /MD /O2 /Ob2  /w /EHa /MP /bigobj   -DCUDA_HAS_FP16=1 -DUSE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD /Z7 /EHa /DNOMINMAX /wd4267 /wd4251 /wd4522 /wd4838 /wd4305 /wd4244 /wd4190 /wd4101 /wd4996 /wd4275 /bigobj -openmp -std:c++14 /showIncludes /Focaffe2\torch\CMakeFiles\torch_python.dir\csrc\autograd\generated\python_nn_functions.cpp.obj /Fdcaffe2\torch\CMakeFiles\torch_python.dir\ /FS -c ..\torch\csrc\autograd\generated\python_nn_functions.cpp 
Microsoft (R) C/C++ Optimizing Compiler Version 19.11.25548.2 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

DWIN32 /D_WINDOWS /GR  /w /EHa /MP /bigobj -openmp -DHAVE_AVX_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION /MD /O2 /Ob2  /w /EHa /MP /bigobj   -DCUDA_HAS_FP16=1 -DUSE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD /Z7 /EHa /DNOMINMAX /wd4267 /wd4251 /wd4522 /wd4838 /wd4305 /wd4244 /wd4190 /wd4101 /wd4996 /wd4275 /bigobj -openmp -std:c++14 /showIncludes /Focaffe2\torch\CMakeFiles\torch_python.dir\csrc\Storage.cpp.obj /Fdcaffe2\torch\CMakeFiles\torch_python.dir\ /FS -c ..\torch\csrc\Storage.cpp 
FAILED: caffe2/torch/CMakeFiles/torch_python.dir/csrc/Storage.cpp.obj  
DWIN32 /D_WINDOWS /GR  /w /EHa /MP /bigobj -openmp -DHAVE_AVX_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION /MD /O2 /Ob2  /w /EHa /MP /bigobj   -DCUDA_HAS_FP16=1 -DUSE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD /Z7 /EHa /DNOMINMAX /wd4267 /wd4251 /wd4522 /wd4838 /wd4305 /wd4244 /wd4190 /wd4101 /wd4996 /wd4275 /bigobj -openmp -std:c++14 /showIncludes /Focaffe2\torch\CMakeFiles\torch_python.dir\csrc\Storage.cpp.obj /Fdcaffe2\torch\CMakeFiles\torch_python.dir\ /FS -c ..\torch\csrc\Storage.cpp 
C:\Users\circleci\project\torch/csrc/autograd/variable.h(347): error C2059: syntax error: 'constant'
C:\Users\circleci\project\torch/csrc/autograd/variable.h(347): error C3805: 'constant': unexpected token, expected either '}' or a ','
C:\Users\circleci\project\torch/csrc/autograd/variable.h(391): error C2589: 'constant': illegal token on right side of '::'
C:\Users\circleci\project\torch/csrc/autograd/variable.h(391): error C2062: type 'unknown-type' unexpected
Microsoft (R) C/C++ Optimizing Compiler Version 19.11.25548.2 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

letest\googletest /DWIN32 /D_WINDOWS /GR  /w /EHa /MP /bigobj -openmp -DHAVE_AVX_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION /MD /O2 /Ob2  /w /EHa /MP /bigobj   -DCUDA_HAS_FP16=1 -DUSE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD /Z7 /EHa /DNOMINMAX /wd4267 /wd4251 /wd4522 /wd4838 /wd4305 /wd4244 /wd4190 /wd4101 /wd4996 /wd4275 /bigobj -openmp -std:c++14 /showIncludes /Fotest_api\CMakeFiles\test_api.dir\modules.cpp.obj /Fdtest_api\CMakeFiles\test_api.dir\ /FS -c ..\test\cpp\api\modules.cpp 
Microsoft (R) C/C++ Optimizing Compiler Version 19.11.25548.2 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

 /D_WINDOWS /GR  /w /EHa /MP /bigobj -openmp -DHAVE_AVX_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION /MD /O2 /Ob2  /w /EHa /MP /bigobj   -DCUDA_HAS_FP16=1 -DUSE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD /Z7 /EHa /DNOMINMAX /wd4267 /wd4251 /wd4522 /wd4838 /wd4305 /wd4244 /wd4190 /wd4101 /wd4996 /wd4275 /bigobj -openmp -std:c++14 /showIncludes /Focaffe2\torch\CMakeFiles\torch_python.dir\csrc\Exceptions.cpp.obj /Fdcaffe2\torch\CMakeFiles\torch_python.dir\ /FS -c ..\torch\csrc\Exceptions.cpp 
Microsoft (R) C/C++ Optimizing Compiler Version 19.11.25548.2 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

albanD committed:
Update on "Change to only warn for custom Function"


As discussed with @gchanan, we do a deprecation cycle for this as some users might rely on the current behavior.
But given that their custom backward is not called in this case (or called with partial gradients), we agree that this is not the expected behavior and will disallow it.

[ghstack-poisoned]

albanD committed:
Update on "Change to only warn for custom Function"


As discussed with @gchanan, we do a deprecation cycle for this as some users might rely on the current behavior.
But given that their custom backward is not called in this case (or called with partial gradients), we agree that this is not the expected behavior and will disallow it.

[ghstack-poisoned]

albanD committed:
Update on "Change to only warn for custom Function"


As discussed with @gchanan, we do a deprecation cycle for this as some users might rely on the current behavior.
But given that their custom backward is not called in this case (or called with partial gradients), we agree that this is not the expected behavior and will disallow it.

[ghstack-poisoned]

albanD committed:
Update on "Change to only warn for custom Function"


As discussed with @gchanan, we do a deprecation cycle for this as some users might rely on the current behavior.
But given that their custom backward is not called in this case (or called with partial gradients), we agree that this is not the expected behavior and will disallow it.

[ghstack-poisoned]

albanD committed:
Update on "Change to only warn for custom Function"


As discussed with @gchanan, we do a deprecation cycle for this as some users might rely on the current behavior.
But given that their custom backward is not called in this case (or called with partial gradients), we agree that this is not the expected behavior and will disallow it.

[ghstack-poisoned]

albanD committed:
Update on "Change to only warn for custom Function"


As discussed with @gchanan, we do a deprecation cycle for this as some users might rely on the current behavior.
But given that their custom backward is not called in this case (or called with partial gradients), we agree that this is not the expected behavior and will disallow it.

Differential Revision: [D19814115](https://our.internmc.facebook.com/intern/diff/D19814115)

[ghstack-poisoned]

albanD committed:
Update on "Change to only warn for custom Function"


As discussed with @gchanan, we do a deprecation cycle for this as some users might rely on the current behavior.
But given that their custom backward is not called in this case (or called with partial gradients), we agree that this is not the expected behavior and will disallow it.

Differential Revision: [D19814115](https://our.internmc.facebook.com/intern/diff/D19814115)

[ghstack-poisoned]

albanD commented:
Closing and merging this into https://github.com/pytorch/pytorch/pull/32839
